commonmark/src/main/java/org/commonmark/parser/InlineParser.java

```diff
@@ -14,4 +13,9 @@
      */
     void parse(String input, Node node);
 
+    /**
+     * @return how many characters were parsed as a reference, {@code 0} if none
+     */
+    int parseReference(String s);
```

Comment:
We could put this method on a separate interface that lives in `.internal` for now, and `ParagraphParser` does an `if (inlineParser instanceof ReferenceParser)` before calling it. That way, we wouldn't have to put it in the API while your extension would still work I think. Thoughts?

That way this PR can go in while I think some more about what a proper `parseReference` should look like.